chatsky/conditions/ml.py

```diff
@@ -0,0 +1,85 @@
+"""
+Conditions
+------------
+
+This module provides condition functions for annotation processing.
+"""
+
+from typing import Optional, List
+
+try:
+    # !!! remove sklearn, use pure python instead
+    from sklearn.metrics.pairwise import cosine_similarity
+
+    sklearn_available = True
+except ImportError:
+    sklearn_available = False
+from chatsky import Context
+from chatsky.conditions.standard import BaseCondition
+from chatsky.ml.models.base_model import ExtrasBaseAPIModel
+
+
+class HasLabel(BaseCondition):
+    """
+    Use this condition, when you need to check, whether the probability
+    of a particular label for the last annotated user utterance surpasses the threshold.
+
+    :param label: String name or a reference to a DatasetItem object, or a collection thereof.
+    :param namespace: Namespace key of a particular model that should detect the dataset_item.
+        If not set, all namespaces will be searched for the required dataset_item.
+    :param threshold: The minimal label probability that triggers a positive response
+        from the function.
+    """
+
+    label: str
+    model_name: str
+    threshold: float = 0.9
+
+    async def call(self, ctx: Context) -> bool:
+        model = ctx.pipeline.models[self.model_name]
+        # Predict labels for the last request
+        # and store them in framework_data with uuid of the model as a key
+        await model(ctx)
```

Comment:
Only call model if `model_id` is not in `models_labels`.
Clear `models_labels` in `pipeline.run` to prevent using labels from previous turn.

Add tests that cover this use case (multiple condition calls to the same model).